src/stk/ea/mutation/mutators/molecule/random_smarts.py

```diff
@@ -0,0 +1,142 @@
+"""
+Random SMARTS
+=============
+
+"""
+
+import numpy as np
+
+
+from .mutator import MoleculeMutator
+from ...records import MutationRecord
+from ....molecule_records import MoleculeRecord
+import rdkit.Chem.AllChem as rdkit
+from .....molecular import BuildingBlock
+
+
+class RandomSmarts(MoleculeMutator):
+    """
+    Substitutes functional groups within building blocks.
+
+    This mutator takes a :class:`.ConstructedMolecule` and substitutes
+    the existing building blocks with ones containing
+    new functionalities, as specified in the SMARTS strings.
+    Atoms in functional groups used to
+    construct the :class: `.ConstructedMolecule` cannot be changed,
+    and will result in an error being raised, unless
+    new functional groups are specified in `replacement_fgs`.
+    Only external functionalities can be replaced.
+
```

Comment:
what are external functionalities?

Reply:
I agree this is ambigious - I meant only atoms not part of the functional groups. I've removed this line